tests/testthat/helper-state.R

```diff
@@ -3,13 +3,19 @@
 # If global settings need to be modified, they should be restored to their
 # original values on exit. This can be achieved with the `on.exit()` base
 # function, or more conveniently with the `withr` package.
+# Note that `globalCallingHandlers` is only available on version of R >= 4.0
+
 testthat::set_state_inspector(function() {
   list(
     attached    = search(),
     connections = getAllConnections(),
     cwd         = getwd(),
     envvars     = Sys.getenv(),
-    handlers    = globalCallingHandlers(),
+    handlers    = if (check_r_version(4, 0 )) {
+      globalCallingHandlers()
+      } else {
+        Sys.getenv("error")
+      },
```

Comment:
```suggestion
    handlers    = if (getRversion() >= "4.0.0") {
      globalCallingHandlers()
      } else {
        Sys.getenv("error")
      },
```

which makes the `check_r_version()` helper unnecessary.

I would even go as far as [wrapping the entire expression in the `if`](https://github.com/epiverse-trace/packagetemplate/blob/ca7cbb8e14745165160db1826da84a303202cf06/tests/testthat/helper-state.R) since as @pearsonca mentioned, we don't expect different results for older versions of R so we maybe don't need to check each time.